### ISSUE TITLE
RocketChat CPU usage 100% after sending specific lines

### ISSUE THREAD

**Issue by NeXT405** [RocketChat/Rocket.Chat#30016]
### Description:

We have found a strange bug(?). 

When we want to send the following lines via RocketChat, the CPU usage is suddenly at 100% (ne the server!) and all clients hang. The server logs nothing out. 

```
java.sql.SQLException: Unable to run insert stmt on object ILPResource [identifier=Kaese-Mostarda-Laugenknopf, href=null, mediaType=application/pdf, pk=0, issue=ILPIssueDef [issueId=FABL202304, reference=RICHM-FABL-2023-DE, pk=46], topic=null]: INSERT INTO `ZILPRESOURCE` (`ZIDENTIFIER` ,`ZHREF` ,`ZRESOURCELOCATION` ,`ZLOCALPATH` ,`ZMEDIATYPE` ,`ZGROUP` ,`ZREFERENCETYPE` ,`ZLASTUPDATE` ,`ZDATA` ,`ZPLAINDATA` ,`ZTITLE` ,`ZSIZEBYTES` ,`ZORDER` ,`ZISSUE` ,`ZTOPIC` ,`ZEXERCISE` ,`ZTOPICDEFINITION` ,`ZEXERCISEDEFINITION` ,`ZASSETDEFINITION` ,`ZLMSCONTENT` ,`Z_ENT` ,`Z_OPT` ) VALUES (?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?)
	at com.j256.ormlite.misc.SqlExceptionUtil.create(SqlExceptionUtil.java:25) ~[ormlite-core-5.7.jar:?]
	at com.j256.ormlite.stmt.mapped.MappedCreate.insert(MappedCreate.java:140) ~[ormlite-core-5.7.jar:?]
	at com.j256.ormlite.stmt.StatementExecutor.create(StatementExecutor.java:465) ~[ormlite-core-5.7.jar:?]
	at com.j256.ormlite.dao.BaseDaoImpl.create(BaseDaoImpl.java:333) ~[ormlite-core-5.7.jar:?]
	at ch.ionesoft.ilp.book.co.n.c(SourceFile:941) ~[?:?]
	at ch.ionesoft.ilp.book.co.n.d(SourceFile:883) ~[?:?]
	at ch.ionesoft.ilp.book.co.n.c(SourceFile:592) ~[?:?]
	at ch.ionesoft.ilp.book.co.n.a(SourceFile:572) ~[?:?]
	at ch.ionesoft.ilp.book.co.n.c(SourceFile:500) ~[?:?]
	at ch.ionesoft.ilp.book.co.n.b(SourceFile:491) ~[?:?]
	at ch.ionesoft.ilp.book.co.n$7.a(SourceFile:328) ~[?:?]
	at ch.ionesoft.ilp.book.co.c$1.a(SourceFile:81) ~[?:?]
	at ch.ionesoft.ilp.book.ia.lima.d.a(SourceFile:326) ~[?:?]
	at ch.ionesoft.ilp.book.co.c.a(SourceFile:76) ~[?:?]
	at ch.ionesoft.ilp.book.co.n.a(SourceFile:992) ~[?:?]
	at ch.ionesoft.ilp.book.co.n.b(SourceFile:325) ~[?:?]
	at ch.ionesoft.ilp.book.co.n.a(SourceFile:103) ~[?:?]
	at ch.ionesoft.ilp.book.mo.ILPEProduct.install(SourceFile:754) ~[?:?]
	at ch.ionesoft.ilp.vc.portalv2.a.a(SourceFile:916) ~[?:?]
	at ch.ionesoft.ilp.vc.portalv2.a.a(SourceFile:748) ~[?:?]
	at ch.ionesoft.ilp.vc.portalv2.a.a(SourceFile:709) ~[?:?]
	at ch.ionesoft.ilp.vc.portalv2.c.d(SourceFile:1052) ~[?:?]
	at ch.ionesoft.ilp.vc.portalv2.c.a(SourceFile:1019) ~[?:?]
	at ch.ionesoft.ilp.vc.portalv2.c$23.run(SourceFile:250) ~[?:?]
	at java.lang.Thread.run(Thread.java:833) ~[?:?]
Caused by: org.sqlite.SQLiteException: [SQLITE_ERROR] SQL error or missing database (table ZILPRESOURCE has no column named ZLMSCONTENT)
	at org.sqlite.core.DB.newSQLException(DB.java:909) ~[sqlite-jdbc-3.15.1.jar:?]
	at org.sqlite.core.DB.newSQLException(DB.java:921) ~[sqlite-jdbc-3.15.1.jar:?]
	at org.sqlite.core.DB.throwex(DB.java:886) ~[sqlite-jdbc-3.15.1.jar:?]
	at org.sqlite.core.NativeDB.prepare_utf8(Native Method) ~[sqlite-jdbc-3.15.1.jar:?]
	at org.sqlite.core.NativeDB.prepare(NativeDB.java:127) ~[sqlite-jdbc-3.15.1.jar:?]
	at org.sqlite.core.DB.prepare(DB.java:227) ~[sqlite-jdbc-3.15.1.jar:?]
	at org.sqlite.core.CorePreparedStatement.<init>(CorePreparedStatement.java:41) ~[sqlite-jdbc-3.15.1.jar:?]
	at org.sqlite.jdbc3.JDBC3PreparedStatement.<init>(JDBC3PreparedStatement.java:30) ~[sqlite-jdbc-3.15.1.jar:?]
	at org.sqlite.jdbc4.JDBC4PreparedStatement.<init>(JDBC4PreparedStatement.java:19) ~[sqlite-jdbc-3.15.1.jar:?]
	at org.sqlite.jdbc4.JDBC4Connection.prepareStatement(JDBC4Connection.java:48) ~[sqlite-jdbc-3.15.1.jar:?]
	at org.sqlite.jdbc3.JDBC3Connection.prepareStatement(JDBC3Connection.java:254) ~[sqlite-jdbc-3.15.1.jar:?]
	at org.sqlite.jdbc3.JDBC3Connection.prepareStatement(JDBC3Connection.java:226) ~[sqlite-jdbc-3.15.1.jar:?]
	at org.sqlite.jdbc3.JDBC3Connection.prepareStatement(JDBC3Connection.java:247) ~[sqlite-jdbc-3.15.1.jar:?]
	at com.j256.ormlite.jdbc.JdbcDatabaseConnection.insert(JdbcDatabaseConnection.java:189) ~[ormlite-jdbc-5.7.jar:?]
	at com.j256.ormlite.stmt.mapped.MappedCreate.insert(MappedCreate.java:92) ~[ormlite-core-5.7.jar:?]
	... 23 more
```

After sending, RocketChat is immediately unavailable. This does not happen if you send it in a multi-line code block. 

It helps only: `docker compose restart`

Here could be the same problem: 
https://github.com/RocketChat/Rocket.Chat/issues/29515
https://github.com/RocketChat/Rocket.Chat/issues/28918

Feel free to ask if you need more informations. 

### Steps to reproduce:

1. Open any chat window
2. Copy-Paste the above lines
3. Press send

### Expected behavior:

The text is successfully transferred 

### Actual behavior:

RoketChat container hangs on Server with 100% CPU usage. 

### Server Setup Information:

- Version of Rocket.Chat Server: 6.2.11 -> But also testet with 6.2.5
- Operating System: Ubuntu 22.04.2 LTS x86_64, 
- Deployment Method: Docker, Engine Version: 24.0.5
- Number of Running Instances: 1
- DB Replicaset Oplog: ?
- NodeJS Version: ?
- MongoDB Version: image: mongo:5.0 🙈 

### Client Setup Information

- Testet sendig this lines on a Macbook air 15" with m2. Also happend on a 13" air with m1. 
- RocketChat 3.9.5

### Additional context

Could not test this on an other instance of RocketChat. 

### Relevant logs:

No logs are provided by RocketChat.

**Comment by Kot-o-pes** [comment]
works like a charm, basically rocket-bomb!

**Comment by NeXT405** [comment]
Interesting. What do you mean by "rocket-bomb"?

**Comment by Kot-o-pes** [comment]
lets say you have a k8s cluster and rocketchat pod not limited in any way, then send this message and all pods on node is starting to throttle really bad, potentially you can just ddos whole server by sending this payload via api to any rocketchat server

**Comment by feuersen** [comment]
Today we hit this bug too. One of our users pasted a C++ stack trace, resulting in the server hanging at 100% CPU load for about 5 minutes.

After finally reproducing this on our test system, I found the problematic pattern:  `_ [] `
The more one repeats this, the longer the server hangs with 100% cpu in main.js, finally crashing with OOM.
E.g. pasting something like this:
```
_ [] _ [] 
_ [] _ [] 
_ [] _ [] 
_ [] _ [] 
_ [] _ [] 
_ [] _ [] 
_ [] _ [] 
_ [] _ [] 
_ [] _ [] 
_ [] _ [] 
_ [] _ [] 
_ [] _ [] 
```


Observed this problem with v6.3.10, v6.3.9 and v6.4.2.

CPU Load during problem:
```
  PID USER      PR  NI    VIRT    RES    SHR S  %CPU %MEM     TIME+ COMMAND
16523 65533     20   0 2386996 921440  26884 R 100.0 23.8   0:39.67 node main.js
 4053 mongod    20   0 3118736   1.2g   9732 S   2.3 31.4   3:42.20 /usr/bin/mongod -f /etc/mongod.conf
...
```

Crash error message:
```
rocketchat-rocketchat-1  | +-------------------------------------------------------+
rocketchat-rocketchat-1  | |                     SERVER RUNNING                    |
rocketchat-rocketchat-1  | +-------------------------------------------------------+
rocketchat-rocketchat-1  | |                                                       |
rocketchat-rocketchat-1  | |  Rocket.Chat Version: 6.3.10                          |
rocketchat-rocketchat-1  | |       NodeJS Version: 14.21.3 - x64                   |
rocketchat-rocketchat-1  | |      MongoDB Version: 4.4.17                          |
rocketchat-rocketchat-1  | |       MongoDB Engine: unknown                         |
rocketchat-rocketchat-1  | |             Platform: linux                           |
rocketchat-rocketchat-1  | |         Process Port: 3000                            |
rocketchat-rocketchat-1  | |             Site URL: https://xoxoxoxo.xoxoxo.xo      |
rocketchat-rocketchat-1  | |     ReplicaSet OpLog: Enabled                         |
rocketchat-rocketchat-1  | |          Commit Hash: 6c79ffc512                      |
rocketchat-rocketchat-1  | |        Commit Branch: HEAD                            |
rocketchat-rocketchat-1  | |                                                       |
rocketchat-rocketchat-1  | +-------------------------------------------------------+
...
rocketchat-rocketchat-1  |
rocketchat-rocketchat-1  | <--- Last few GCs --->
rocketchat-rocketchat-1  |
rocketchat-rocketchat-1  | [219:0x6101870]   453519 ms: Scavenge 1359.2 (1397.3) -> 1343.2 (1397.3) MB, 1.4 / 0.4 ms  (average mu = 0.907, current mu = 0.836) allocation failure
rocketchat-rocketchat-1  | [219:0x6101870]   453556 ms: Scavenge 1359.2 (1397.3) -> 1343.2 (1397.3) MB, 1.5 / 0.4 ms  (average mu = 0.907, current mu = 0.836) allocation failure
rocketchat-rocketchat-1  | [219:0x6101870]   453592 ms: Scavenge 1359.2 (1397.3) -> 1343.2 (1397.3) MB, 1.4 / 0.4 ms  (average mu = 0.907, current mu = 0.836) allocation failure
rocketchat-rocketchat-1  |
rocketchat-rocketchat-1  |
rocketchat-rocketchat-1  | <--- JS stacktrace --->
rocketchat-rocketchat-1  |
rocketchat-rocketchat-1  | FATAL ERROR: invalid array length Allocation failed - JavaScript heap out of memory
rocketchat-rocketchat-1  |  1: 0xa3ad50 node::Abort() [node]
rocketchat-rocketchat-1  |  2: 0x970199 node::FatalError(char const*, char const*) [node]
rocketchat-rocketchat-1  |  3: 0xbba90e v8::Utils::ReportOOMFailure(v8::internal::Isolate*, char const*, bool) [node]
rocketchat-rocketchat-1  |  4: 0xbbac87 v8::internal::V8::FatalProcessOutOfMemory(v8::internal::Isolate*, char const*, bool) [node]
rocketchat-rocketchat-1  |  5: 0xd76ea5  [node]
rocketchat-rocketchat-1  |  6: 0xd4f6d5  [node]
rocketchat-rocketchat-1  |  7: 0xed93be  [node]
rocketchat-rocketchat-1  |  8: 0xed950d  [node]
rocketchat-rocketchat-1  |  9: 0x1077ac3 v8::internal::Runtime_GrowArrayElements(int, unsigned long*, v8::internal::Isolate*) [node]
rocketchat-rocketchat-1  | 10: 0x1449379  [node]
rocketchat-rocketchat-1  | Tried 15 times. Waiting 5 secs...
rocketchat-rocketchat-1  | bash: line 1:   219 Aborted                 (core dumped) node main.js
```

**Comment by froque** [comment]
this seems connected to `packagingData` concept in logback. 
https://logback.qos.ch/manual/configuration.html#packagingData

disabling this might help

**Comment by MartinSchoeler** [comment]
Hey, this PR fixes the issue https://github.com/RocketChat/fuselage/pull/1231
Expect the fix to come in version 6.5.0
Thanks!
